tests/manage/pv_services/pvc_resize/test_resource_deletion_during_pvc_expansion.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import pytest
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version, ManageTest, tier4, tier4b, ignore_leftover_label
+)
+from tests import disruption_helpers
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@tier4b
+@skipif_ocs_version('<4.5')
+@ignore_leftover_label(constants.drain_canary_pod_label)
+@pytest.mark.parametrize(
+    argnames='resource_to_delete',
+    argvalues=[
+        pytest.param(
+            'mgr', marks=pytest.mark.polarion_id('OCS-2224')
+        ),
+        pytest.param(
+            'osd', marks=pytest.mark.polarion_id('OCS-2225')
+        ),
+        pytest.param(
+            'rbdplugin', marks=pytest.mark.polarion_id('OCS-2226')
+        ),
+        pytest.param(
+            'cephfsplugin', marks=pytest.mark.polarion_id('OCS-2227')
+        ),
+        pytest.param(
+            'rbdplugin_provisioner', marks=pytest.mark.polarion_id('OCS-2228')
+        ),
+        pytest.param(
+            'cephfsplugin_provisioner',
+            marks=pytest.mark.polarion_id('OCS-2229')
+        )
+    ]
+)
+class TestResourceDeletionDuringPvcExpansion(ManageTest):
+    """
+    Tests to verify PVC expansion will be success even if rook-ceph, csi pods
+    are re-spun during the expansion
+
+    """
+    @pytest.fixture(autouse=True)
+    def setup(self, create_pvcs_and_pods):
+        """
+        Create PVCs and pods
+
+        """
+        self.pvcs, self.pods = create_pvcs_and_pods(pvc_size=10, dc_for_rwx=2)
+
+    def test_resource_deletion_during_pvc_expansion(self, resource_to_delete):
+        """
+        Verify PVC expansion will succeed when rook-ceph, csi pods are re-spun
+        during expansion
+
+        """
+        pvc_size_expanded = 30
+        executor = ThreadPoolExecutor(max_workers=len(self.pvcs))
+        disruption_ops = disruption_helpers.Disruptions()
+
+        # Run IO to fill some data
+        log.info(
+            "Running IO on all pods to fill some data before PVC expansion."
+        )
+        for pod_obj in self.pods:
+            storage_type = (
+                'block' if pod_obj.pvc.volume_mode == 'Block' else 'fs'
+            )
+            pod_obj.run_io(
+                storage_type=storage_type, size='4G', io_direction='write',
+                runtime=30, fio_filename=f'{pod_obj.name}_f1'
```

Comment:
add `rate=100` so the filling will be faster

Reply:
Shouldn't this be in format rate=int[,int][,int] ?
For specifying writes only rate=,100m
Reference - https://fio.readthedocs.io/en/latest/fio_doc.html

Reply:
See the example from this link to fio docs here:

`For example, using rate=1m,500k would limit reads to 1MiB/sec and writes to 500KiB/sec.`
![Screenshot from 2020-07-06 22-31-32](https://user-images.githubusercontent.com/6973523/86632818-8a28eb00-bfd8-11ea-9548-6650c3be1dda.png)

We are also using `rate` in the same matter in https://github.com/red-hat-storage/ocs-ci/blob/4d6a077b83e53b78327b42e0791a677f267fa995/ocs_ci/ocs/cluster_load.py#L238

Reply:
Added rate='100M'

Reply:
When rate='100M' , I see lot of MDS slow response messages and client IO will fill up the network bandwidth. So reducing the rate to 10M.